core/src/main/java/org/infinispan/partitionhandling/impl/PartitionHandlingManager.java

```diff
@@ -27,6 +27,14 @@
 
    void checkBulkRead();
 
+   /**
+    * Unlike in {@link #checkRead(Object)}, here we ignore the availability status and we only fail the operation if
+    * _all_ owners have left the cluster.
+    *
+    * @throws org.infinispan.partitionhandling.AvailabilityException if all owners of the keys are missing
+    */
+   void checkForMissingKeys(Collection<?> expected, Collection<?> actual);
```

Comment:
Have you merely refactored the code or did you manage to understand it? Recently I got under the impression that the code has piled up over time and we should just throw `AvailabilityException` without the check at all.